libbeat/autodiscover/providers/kubernetes/kubernetes_test.go

```diff
@@ -104,20 +100,20 @@ func TestNewLeaderElectionManager(t *testing.T) {
 
 	lease := createLease()
 	// create the lease that leader election will be using
-	err := applyLease(client, lease, true)
+	err := applyLease(client, lease)
```

Comment:
So we removed the channel overall and replaced it with just the eventId. Not against it but was sth that was working and had no complains. Any reason?

Reply:
@gizas : Please note this is just a backport PR. 
The changes were done by @MichaelKatsoulis in this main [PR](https://github.com/elastic/beats/pull/43540) for K8s related file.

Reply:
Yes see the comment above: https://github.com/elastic/beats/pull/44036#discussion_r2063429114 
It was hidden after resolving the comment. Please neglect